src/JointTrajectoryControllerBase.cpp

```diff
@@ -202,10 +211,22 @@ void JointTrajectoryControllerBase::updateStep(const ros::Time& time,
     trajectory->evaluateDerivative(timeFromStart, 2, mDesiredAcceleration);
 
     // TODO: Check path constraints.
-    // TODO: Check goal constraints.
+
+    // Check goal constraints.
+    bool goalConstraintsSatisfied = true;
+    for (const auto& pair : mGoalConstraints) {
+      auto dof = mSkeleton->getDof(pair.first);
```

Comment:
It seems that we're using `mSkeleton` here rather than `mControlledSkeleton`. Is this because `MetaSkeleton::getDof(std::string name)` isn't defined by DART?

I think that this means that `index` is now an index into the DOFs of the underlying `Skeleton`, not the `MetaSkeleton` that we're actually controlling. This seems like a bug since `mDesiredPosition` and `mActualPosition` are constructed from `mControlledSkeleton`, so it might throw an error if there are extra DOFs in the `Skeleton` that aren't part of the `MetaSkeleton`.

One alternative is to invert this by iterating through the `MetaSkeleton` DOFs (by index) and then checking whether the DOF name is in `mGoalConstraints`.

Reply:
oh, good point. I'll need to take a closer look.